client_cli/src/main.rs

```diff
@@ -420,6 +422,29 @@ mod domain {
             Ok(())
         }
     }
+
+    /// Transfer a domain between accounts
+    #[derive(Debug, StructOpt)]
+    pub struct Transfer {
+        /// Domain name as double-quited string
+        #[structopt(short, long)]
+        pub id: DomainId,
+        /// Account from which to transfer (in form `name@domain_name')
+        #[structopt(short, long)]
+        pub from: AccountId,
+        /// Account from which to transfer (in form `name@domain_name')
+        #[structopt(short, long)]
+        pub to: AccountId,
+    }
+
+    impl RunArgs for Transfer {
+        fn run(self, context: &mut dyn RunContext) -> Result<()> {
+            let Self { id, from, to } = self;
+            let transfer_domain = TransferExpr::new(from, id, to);
+            submit([transfer_domain], UnlimitedMetadata::new(), context)
```

Comment:
So that's the place where custom metadata could be.

Yes, I don't know the exact reasons why do we even have metadata for transactions. But I can imagine that for custom chains it could be usefull to attach some additional info for transaction. I.e. it can be checked insise runtime executor. Or analyzed by some off-chain listener.

Reply:
Agreed, this is a very good use case where it is needed.